### PR TITLE
Benchmark http client

### PR DESCRIPTION
**Pull request by alsamylkin**
Add a benchmark test for the http client filters
Increase coverage for tracing errors
Ensure all filters are copied on subsequent calls in client

### CODE REVIEW

File: modules/uhttp/client/client_filters_test.go [hunk 1]
```diff
@@ -121,6 +121,44 @@ func TestExecutionChainFilters_AuthContextPropagationFailure(t *testing.T) {
...
+	for name, filter := range testCases {
+		op := func(tracer opentracing.Tracer) {
+			tr := &shadowTracer{
+				tracer,
+				func(sm opentracing.SpanContext, format interface{}, carrier interface{}) error {
+					return errors.New("Very bad tracer")
+				},
+				nil,
+			}
+			opentracing.InitGlobalTracer(tr)
```
Comment (anuptalwalkar) [inline modules/uhttp/client/client_filters_test.go]: This call returns a triplet - tracer, closer, error. You want to check for err, and close?

File: modules/uhttp/client/client_filters_benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber-go/tally"
+	jconfig "github.com/uber/jaeger-client-go/config"
+)
+
+// BenchmarkClientFilters/empty-8         	  500000	      3517 ns/op	     256 B/op	       2 allocs/op
+// BenchmarkClientFilters/tracing-8       	   20000	     64421 ns/op	    1729 B/op	      29 allocs/op
+// BenchmarkClientFilters/auth-8          	   50000	     36574 ns/op	     728 B/op	      16 allocs/op
+// BenchmarkClientFilters/default-8       	   10000	    104374 ns/op	    2275 B/op	      43 allocs/op
```
Comment (anuptalwalkar) [inline modules/uhttp/client/client_filters_benchmark_test.go]: Can you add these to uhttp doc?

File: modules/uhttp/client/client_filters_test.go (hunk 1 above)
Comment (alsamylkin) [inline modules/uhttp/client/client_filters_test.go]: I am using a different function:
https://github.com/opentracing/opentracing-go/blob/master/globaltracer.go#L30

Comment (anuptalwalkar) [inline modules/uhttp/client/client_filters_test.go]: Ah. Got it.

**Pull request merged**